crates/bevy_reflect/bevy_reflect_derive/src/impls/enums.rs

```diff
@@ -262,126 +261,99 @@ struct EnumImpls {
 fn generate_impls(reflect_enum: &ReflectEnum, ref_index: &Ident, ref_name: &Ident) -> EnumImpls {
     let bevy_reflect_path = reflect_enum.meta().bevy_reflect_path();
 
-    let mut variant_info: Vec<proc_macro2::TokenStream> = Vec::new();
-    let mut enum_field: Vec<proc_macro2::TokenStream> = Vec::new();
-    let mut enum_field_at: Vec<proc_macro2::TokenStream> = Vec::new();
-    let mut enum_index_of: Vec<proc_macro2::TokenStream> = Vec::new();
-    let mut enum_name_at: Vec<proc_macro2::TokenStream> = Vec::new();
-    let mut enum_field_len: Vec<proc_macro2::TokenStream> = Vec::new();
-    let mut enum_variant_name: Vec<proc_macro2::TokenStream> = Vec::new();
-    let mut enum_variant_type: Vec<proc_macro2::TokenStream> = Vec::new();
+    let mut variant_info = Vec::new();
+    let mut enum_field = Vec::new();
+    let mut enum_field_at = Vec::new();
+    let mut enum_index_of = Vec::new();
+    let mut enum_name_at = Vec::new();
+    let mut enum_field_len = Vec::new();
+    let mut enum_variant_name = Vec::new();
+    let mut enum_variant_type = Vec::new();
 
     for variant in reflect_enum.active_variants() {
         let ident = &variant.data.ident;
         let name = ident.to_string();
         let unit = reflect_enum.get_unit(ident);
 
-        match &variant.fields {
+        fn for_fields(
+            fields: &[StructField],
+            mut generate_for_field: impl FnMut(usize, usize, &StructField) -> proc_macro2::TokenStream,
+        ) -> (usize, Vec<proc_macro2::TokenStream>) {
+            let mut constructor_argument = Vec::new();
+            let mut reflect_idx = 0;
+            for field in fields.iter() {
+                if field.attrs.ignore {
+                    // Ignored field
+                    continue;
+                }
+                constructor_argument.push(generate_for_field(reflect_idx, field.index, field));
```

Comment:
I use `StructField.index` here instead of an iteration index, similarly to the rest of the derive code.